src/Sound/Tidal/ParseBP.hs

```diff
@@ -560,22 +560,35 @@ pRational :: MyParser (TPat Rational)
 pRational = wrapPos $ TPat_Atom Nothing <$> pRatio
 
 pRatio :: MyParser Rational
-pRatio = do 
+pRatio = do
   s <- sign
-  r <- do n <- try intOrFloat
+  r <- do n <- try (try pFloat <|> pInteger)
```

Comment:
At this point `try pFloat <|> pInteger` could be the implementation of `intOrFloat`.
If so, everything will work as expected, except a test: 
https://github.com/tidalcycles/Tidal/blob/12c40a5b6d9fafcd3b895d9b90de1ac97483cbbf/test/Sound/Tidal/UITest.hs#L139

The correct event is:
`Event (Context [((1,1),(4,1))]) Nothing (Arc 0 0.5) (6 :: Int), Event (Context [((4,1),(5,1))]) Nothing (Arc 0.5 1) (0 :: Int)`

This will avoid duplication.

Reply:
Yes you're right, we should probably replace `inOrFloat`. 
I don't really understand why this test fails, and I also don't see the difference between the events in the test file and the events you posted ? What am I missing?

Reply:
The test is wrong, it expect the context of the event to be from the character 1 to 4 of the pattern, but in reality is 1 to 3.
Whops, I pasted the wrong event! this is the right one:
`Event (Context [((1,1),(3,1))]) Nothing (Arc 0 0.5) (6 :: Int), Event (Context [((4,1),(5,1))]) Nothing (Arc 0.5 1) (0 :: Int)`

Reply:
Ah, makes sense, thanks! Just added a new commit.